community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/isGraphKernelResultValue.scala

```diff
@@ -19,13 +19,9 @@
  */
 package org.neo4j.cypher.internal
 
-import org.neo4j.graphdb.{Path, Node, Relationship}
+import org.neo4j.graphdb.{Node, Path, Relationship}
 
 object isGraphKernelResultValue extends (Any => Boolean) {
-  override def apply(v: Any): Boolean = v match {
-    case node: Node => true
-    case rel: Relationship => true
-    case path: Path => true
-    case _ => false
-  }
+
+  override def apply(v: Any): Boolean = v.isInstanceOf[Node] || v.isInstanceOf[Relationship] || v.isInstanceOf[Path]
```

Comment:
I used `Entity` and that was actually slower than doing the check twice. Not sure why though.